nilearn/_utils/data_gen.py

```diff
@@ -147,6 +156,8 @@ def generate_regions_ts(
         end = int(min(n_features, boundaries[n + 1] + overlap_end))
         win = scipy.signal.get_window(window, end - start)
         win /= win.mean()  # unity mean
+        if negative_regions and rand_gen.choice(a=[True, False]):
+            win = -1 * win
```

Comment:
I've added `negative_regions` parameter to be able to generate negative valued regions as well. I cannot judge if there is an inconvenience due to that.